metaseq/data/streaming_src_tgt_dataset.py

```diff
@@ -207,3 +215,49 @@ def yield_src_tgt_single_sentences_pad_8(iterable, block_size, drop_last, paddin
             "src_block": torch.cat(cur_src_block),
             "tgt_block": torch.cat(cur_tgt_block),
         }
+
+
+def yield_src_tgt_single_sentences_withneg_pad_8(
+    iterable, block_size, drop_last, padding_idx
+):
+    """
+    Mimics sample-break-mode eos_pad_8 with additionally adding negative candidates.
+    """
+    counter = 0
+    for all_src, all_tgt in iterable:
+        if not isinstance(all_src, tuple):
+            all_src = (all_src,)
+            all_tgt = (all_tgt,)
+        for idx, (src, tgt) in enumerate(zip(all_src, all_tgt)):
+            cur_src_block = []
+            cur_src_block_ids = []
+            cur_tgt_block = []
+            if src.numel() > block_size:
+                # truncate right side
+                # TODO: Enable left side truncation
+                src = src[:block_size]
+                tgt = tgt[:block_size]
+
+            cur_src_block.append(src)
+            cur_src_block_ids.append(counter)
+            cur_tgt_block.append(tgt)
+            counter += 1
+
+            # We round up to a multiple of 8 + 1, because later on
+            # one element is removed for src/target tensor creation
+            # which brings it back to a multiple of 8. block_size is
+            # already passed with + 1 included.
+            # cur_block_remain = int(min(math.pow(2, math.ceil(math.log(src.numel(), 2))) + 1, block_size))
+            cur_block_remain = min(int(math.ceil(src.numel() / 8)) * 8 + 1, block_size)
+            cur_block_remain -= src.numel()
+            padding = cur_src_block[-1].new_full((cur_block_remain,), padding_idx)
+            cur_src_block.append(padding)
+            cur_tgt_block.append(padding)
+
+            yield {
+                "ids": torch.LongTensor(cur_src_block_ids),
+                "src_block": torch.cat(cur_src_block),
+                "tgt_block": torch.cat(cur_tgt_block),
+                "is_positive": torch.tensor([1 if idx == 0 else 0]).bool(),
```

Comment:
This function is a duplicate of yield_src_tgt_single_sentences_pad_8 with two additional fields in each returned element, right? Perhaps be more explicit about this in the description in line 224. Alternatively, can this be implemented in yield_src_tgt_single_sentences_pad_8 with a boolean argument to control whether to add the two additional fields?

Reply:
removed this module